rotkehlchen/inquirer.py

```diff
@@ -292,27 +270,89 @@ def find_usd_price(asset: Asset, ignore_cache: bool = False) -> Price:
 
             try:
                 price = oracle_instance.query_current_price(
-                    from_asset=asset,
-                    to_asset=A_USD,
+                    from_asset=from_asset,
+                    to_asset=to_asset,
                 )
             except (PriceQueryUnsupportedAsset, RemoteError) as e:
                 log.error(
-                    f'Current price oracle {oracle} failed to request '
-                    f'USD price for {asset.identifier} due to: {str(e)}.',
+                    f'Current price oracle {oracle} failed to request {to_asset.identifier} '
+                    f'price for {from_asset.identifier} due to: {str(e)}.',
                 )
                 continue
 
             if price != Price(ZERO):
                 log.debug(
-                    f'Current price oracle {oracle} got USD price',
-                    asset=asset,
+                    f'Current price oracle {oracle} got {to_asset.identifier} price',
+                    from_asset=from_asset,
+                    to_asset=to_asset,
                     price=price,
                 )
                 break
 
-        Inquirer._cached_current_price[asset] = CachedPriceEntry(price=price, time=ts_now())
+        Inquirer._cached_current_price[cache_key] = CachedPriceEntry(price=price, time=ts_now())
         return price
 
+    @staticmethod
+    def find_price(
+            from_asset: Asset,
+            to_asset: Asset,
+            ignore_cache: bool = False,
+    ) -> Price:
+        """Returns the current price of 'from_asset' in 'to_asset' valuation.
+        NB: prices for special symbols in any currency but USD are not supported.
+
+        Returns Price(ZERO) if all options have been exhausted and errors are logged in the logs
+        """
+        if from_asset == to_asset:
+            return Price(FVal('1'))
+
+        instance = Inquirer()
+        if to_asset == A_USD:
+            return instance.find_usd_price(asset=from_asset, ignore_cache=ignore_cache)
+
+        cache_key = (from_asset, to_asset)
+        if ignore_cache is False:
+            cache = instance._cached_current_price.get(cache_key, None)
```

Comment:
As mentioned above this logic should be abstracted in the is_cache_valid() function